sdk/core/src/policies/telemetry_policy.rs

```diff
@@ -0,0 +1,131 @@
+use crate::policies::{Policy, PolicyResult};
+use crate::{Context, Request, Response};
+
+use http::{header::USER_AGENT, HeaderValue};
+use std::env::consts::{ARCH, OS};
+use std::sync::Arc;
+
+#[derive(Clone, Debug, Default)]
+pub struct TelemetryOptions {
+    pub application_id: Option<String>,
+}
+
+#[derive(Clone, Debug)]
+pub struct TelemetryPolicy {
+    header: String,
+}
+
+/// Sets the User-Agent header with useful information in a typical format for Azure SDKs.
+///
+/// Client libraries should create a `TelemetryPolicy` using `option_env!()` like so:
+/// ```
+/// use azure_core::policies::{TelemetryOptions, TelemetryPolicy};
+/// let policy = TelemetryPolicy::new(option_env!("CARGO_PKG_NAME"), option_env!("CARGO_PKG_VERSION"), &TelemetryOptions::default());
+/// ```
+impl<'a> TelemetryPolicy {
+    pub fn new(
+        crate_name: Option<&'a str>,
+        crate_version: Option<&'a str>,
+        options: &TelemetryOptions,
+    ) -> Self {
+        Self::new_with_rustc_version(
+            crate_name,
+            crate_version,
+            option_env!("AZSDK_RUSTC_VERSION"),
+            options,
+        )
+    }
+
+    fn new_with_rustc_version(
+        crate_name: Option<&'a str>,
+        crate_version: Option<&'a str>,
+        rustc_version: Option<&'static str>,
```

Comment:
This can also be `Option<&'a str>`. It doesn't need to have a static lifetime since you're just formatting it into a string.